fastplotlib/layouts/_gridplot.py

```diff
@@ -25,128 +25,197 @@ def to_array(a) -> np.ndarray:
 
             This allows custom assignment of controllers
 
-            | Example:
-            | unique controllers for a 2x2 gridplot: np.array([[0, 1], [2, 3]])
-            | same controllers for first 2 plots and last 2 plots: np.array([[0, 0, 1], [2, 3, 3]])
+            | Example with integers:
+            | sync first 2 plots, and sync last 2 plots: [[0, 0, 1], [2, 3, 3]]
+            | Example with str subplot names:
+            | list of lists of subplot names, each sublist is synced: [[subplot_a, subplot_b], [subplot_f, subplot_c]]
+            | this syncs subplot_a and subplot_b together; syncs subplot_e and subplot_c together
 
         canvas: WgpuCanvas, optional
             Canvas for drawing
 
         renderer: pygfx.Renderer, optional
             pygfx renderer instance
 
-        size: (int, int)
+        size: (int, int), optional
             starting size of canvas, default (500, 300)
 
+        names: list or array of str, optional
+            subplot names
         """
 
         self.shape = shape
 
+        if names is not None:
+            if len(list(chain(*names))) != self.shape[0] * self.shape[1]:
+                raise ValueError("must provide same number of subplot `names` as specified by gridplot shape")
+
+            self.names = to_array(names).reshape(self.shape)
+        else:
+            self.names = None
+
         canvas, renderer = make_canvas_and_renderer(canvas, renderer)
 
         if isinstance(cameras, str):
-            if cameras not in valid_cameras:
-                raise ValueError(
-                    f"If passing a str, `cameras` must be one of: {valid_cameras}"
-                )
             # create the array representing the views for each subplot in the grid
             cameras = np.array([cameras] * self.shape[0] * self.shape[1]).reshape(
                 self.shape
             )
 
-        if isinstance(controllers, str):
-            if controllers == "sync":
-                controllers = np.zeros(
-                    self.shape[0] * self.shape[1], dtype=int
-                ).reshape(self.shape)
+        # list -> array if necessary
+        cameras = to_array(cameras).reshape(self.shape)
 
-        if controllers is None:
-            controllers = np.arange(self.shape[0] * self.shape[1]).reshape(self.shape)
+        if cameras.shape != self.shape:
+            raise ValueError("Number of cameras does not match the number of subplots")
+
+        # create the cameras
+        self._cameras = np.empty(self.shape, dtype=object)
+        for i, j in product(range(self.shape[0]), range(self.shape[1])):
+            self._cameras[i, j] = create_camera(camera_type=cameras[i, j])
+
+        if controller_ids is None:
+            # individual controller for each subplot
+            controller_ids = np.arange(self.shape[0] * self.shape[1]).reshape(self.shape)
+
+        elif isinstance(controller_ids, str):
```

Comment:
Are there any other options besides `"sync"`...if not should we throwing an error if `str` passed in not valid?

Reply:
yes you're right, we can have more extensive checks